### PR TITLE
Add a new submodule for running rustfix on

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: tests/fixtures.rs [hunk 1]
```diff
@@ -11,10 +11,24 @@ fn fixtures() {
...
+                use std::io::Read;
+                let mut contents = String::new();
+                file.read_to_string(&mut contents)?;
+                Ok(contents)
+            });
+        let dir = if let Ok(path) = file {
+            fixture_path.join(path.trim())
+        } else {
+            fixture_path.clone()
+        };
```
Comment (killercup) [inline tests/fixtures.rs]: So, we now have files that tell us which directory to look into for applying fixtures? Why not allow multiple sub paths to check? I'd probably start refactoring the whole nested for loop business with a bunch of flat_maps now going from crate folders to tests.

Comment (oli-obk) [inline tests/fixtures.rs]: yea that's the next step :)

File: tests/tests/trust-dns/yolo/diff.diff [hunk 2]
```diff
@@ -0,0 +1,1384 @@
...
+diff --git a/client/src/op/response_code.rs b/client/src/op/response_code.rs
+index 7fb567e..417bed7 100644
+--- a/client/src/op/response_code.rs
++++ b/client/src/op/response_code.rs
+@@ -132,7 +132,7 @@ impl ResponseCode {
+ 
+     /// returns the high 12 bits for the edns portion of the response code
+     pub fn high(&self) {
+-        (u16::from(*self) & 0x0FF0) >> 4;
++        (u16::from(*self) & 0x0FF0);4;
```
Comment (oli-obk) [inline tests/tests/trust-dns/yolo/diff.diff]: lol

**Comment by oli-obk** [comment]
I have no clue how the tests are failing... they work locally...

**Comment by killercup** [comment]
I this because of different versions of diff maybe? (If you're using macos, you might have bsd diff and travis is using gnu diff or something like that.) I haven't read the output, though. (How would we even debug that? A diff of a diff? :D)

Also, for checking the output it might be nice to disable colored output. Don't know if the crate we are using has an override for that, though.

**Comment by oli-obk** [comment]
I like it with the colors. I can do `cat output.txt` and see what it looks like.

**Comment by oli-obk** [comment]
It might also be that we don't get things in the same order every time we run rustfix. I'll investigate

**Comment by oli-obk** [comment]
whoops

**Pull request closed without merging**